### ISSUE TITLE
directory types and file names

### ISSUE THREAD

**Issue by jayayy** [qubvel/segmentation_models#226]
i have the following dataset structure:

train test val folders -> each containing img and mask folder -> each image (.jpg) in img folder has a corresponding mask (.png) in mask folder with "-mask" added in the filename.
eg.: E1245W.jpg has mask name E1245W-mask.png

i am having difficulty using these images to flow in model to get trained. any guidance here would be appreciated.

**Comment by JordanMakesMaps** [comment]
Might I recommend looking at #206? It might be able to answer some of your questions.

**Comment by jayayy** [comment]
thanks it was immensely similar. but could you help with dataframe structure?

is it similar to lisdir(path) for img and mask concaenated?

**Comment by JordanMakesMaps** [comment]
DataFrame comes from [Pandas](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html) where there are two columns (like lists), one contains all the absolute paths for the images, and the other contains the absolute paths for the masks.

But if you just wanted to pass in two lists containing the image and mask paths you could adapt the code so it would be something like this:

```python
class DataGenerator(keras.utils.Sequence):
    
    def __init__(self, images, masks, batch_size, augment, n_classes = 8):
        self.image_path = images
        self.mask_path = masks
        self.batch_size = batch_size
        self.n_classes = n_classes
        self.augment = augment
        self.index = 0
     
     def __check_index__(self):
         return self.index

     def __update_index__(self):
         self.index = self.index + self.batch_size
        
    # Steps per epoch    
    def __len__(self):
        
        # maybe something to confirm equal lengths
        assert len(self.image_path) == len(self.mask_path)

        return len(self.image_path) // self.batch_size
    
    # Shuffles the images and masks **together**
    def on_epoch_end(self):
        
        combined = list(zip(self.image_path, self.mask_path))

        # import random
        random.shuffle(combined)

        self.image_path[:], self.mask_path[:] = zip(*combined)
    
    
    # Generates data, feeds to training
    def __getitem__(self):
        
        index = self.__check_index__()
        
        global preprocess_input
        
        processed_images = []
        processed_masks = []
        
        for _ in range(self.batch_size):

            the_image = io.imread(self.image_path[_])
            
            # This assumes that the masks are a .npy file, change as needed
            the_mask = np.load(self.mask_path[_]).astype('uint8');
            one_hot_mask = one_hot_encoder(the_mask, self.n_classes)
            
            if(self.augment):
                
                processed_image = augs_for_images(image = the_image)
                processed_mask = augs_for_masks(image = one_hot_mask)
         
            else:
                # Still resizing and then random cropping, but no augmentations   
                processed_image = resize_for_images(image = the_image)
                processed_mask = resize_for_masks(image = one_hot_mask)

            processed_images.append(preprocess_input(processed_image))
            processed_masks.append(processed_mask)

                
        batch_x = np.array( processed_images )
        batch_y = np.array( processed_masks )
 
        self.__update__index()
        
        return (batch_x, batch_y)


...

images = glob.glob("folder//holding//images//*.png")
masks = glob.glob("folder//holding//masks//*.npy")

data_generator = DataGenerator(images, masks, batch_size = 1, augment = True, n_classes = 8)

```
You'll definitely need to work with the code, I just edited this on a notepad and haven't ran it to confirm that it works.

**Comment by jayayy** [comment]
This is amazing. Thank you for your responses. Closing the issue now.

thanks again, would communicate again if anything comes up.
